github-upload-action-secrets

```diff
@@ -146,7 +154,28 @@ def main():
         secret_name = opts.ssh_keygen
         public, private = generate_ssh_key(pubkey)
         upload(secrets_path, secret_name, private)
-        sys.stdout.buffer.write(public + b'\n')
+
+        if opts.deploy_to:
+            title = f"{secrets_path}/{secret_name}"
+            keys_path = f"/repos/{opts.deploy_to}/keys"
```

Comment:
The deploy code is sufficiently large to better be moved into a separate function.

Reply:
Ya..... The problem is the `opts` thing.  I might just bump that to be global...

This is the same reason `upload()` is nested and not a proper function of its own.

Reply:
But you could just pass the necessary opts.* values to the function?

Reply:
- added `upload_deploy_key()` as function
 - renamed `upload()` to `upload_secret()` and made it a proper function
 - added `api`, `pubkey`, `opts`, and about 12 other arguments ;)

Starting to think this might benefit from a small utility class....